### PR TITLE
feat: BESC-00 Implement get book list api

### PR DESCRIPTION
**Pull request by phucnguyen97**

### CODE REVIEW

File: src/apps/book/dtos/get-book-list-dto.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+  @IsOptional()
+  @IsNumber()
+  skip: number;
+
+  @IsOptional()
+  sort: SortType;
+
+  @IsOptional()
+  title: string;
```
Comment (quangtran85) [inline src/apps/book/dtos/get-book-list-dto.ts]: `title?: string;`

File: src/apps/book/services/book.service.ts [hunk 2]
```diff
@@ -52,4 +68,33 @@ export class BookService {
       },
     };
   }
+
+  async getBookList(
+    data: GetBookListInput,
+  ): Promise<ResourcesPaginateOutput<GetBookListOutput>> {
+    const { title } = data;
+    console.log('data', data);
```
Comment (quangtran85) [inline src/apps/book/services/book.service.ts]: Pls remove all of debug logs `console.log`

File: src/apps/book/dtos/get-book-list-dto.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import { IsOptional, IsNumber } from 'class-validator';
+import { SortType } from '@shared/type';
+
+export class GetBookListDto {
```
Comment (quangtran85) [inline src/apps/book/dtos/get-book-list-dto.ts]: It should extend `PaginateDto` class from `src/shared/dtos/paginate.dto.ts`

File: src/apps/book/dtos/get-book-list-dto.ts (hunk 1 above)
Comment (quangtran85) [reply src/apps/book/dtos/get-book-list-dto.ts:17]: Btw, named as `keyword` might be better to extends.

**Review by quangtran85 — COMMENTED** [review]
@phucnguyen97 I have some comments, pls check.

**Comment by phucnguyen97** [comment]
> @phucnguyen97 I have some comments, pls check.

I have fixed, please help me review

**Comment by quangtran85** [comment]
@phucnguyen97 Why the code changed of this PR https://github.com/quangtran85/serverless-tutorial/pull/6/files is appeared in the diff?

**Review by quangtran85 — APPROVED** [review]
LGTM

**Pull request merged**